Runtimes/Core/cmake/caches/Vendors/Apple/Darwin.toolchain.cmake

```diff
@@ -0,0 +1,116 @@
+# Toolchain configuration when compiling for Darwin Platforms
+set(CMAKE_SYSTEM_NAME Darwin)
+set(CMAKE_Swift_COMPILER_WORKS YES)
+set(CMAKE_C_COMPILER_WORKS YES)
+set(CMAKE_CXX_COMPILER_WORKS YES)
+
+find_program(XCRUN_EXECUTABLE NAMES "xcrun" REQUIRED)
+
+if(NOT CMAKE_OSX_SYSROOT)
+  if(DEFINED ENV{SDKROOT})
+    set(sdk_name $ENV{SDKROOT})
+  else()
+    message(FATAL_ERROR "SDKROOT environment variable not set")
+  endif()
+
+  execute_process(COMMAND "${XCRUN_EXECUTABLE}" --show-sdk-path -sdk ${sdk_name}
+    OUTPUT_VARIABLE SDKROOT
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if(NOT EXISTS ${SDKROOT})
+    message(FATAL_ERROR "SDKROOT could not be detected!")
+  endif()
+
+  message(STATUS "Using SDKROOT: ${SDKROOT}")
+  set(CMAKE_OSX_SYSROOT "${SDKROOT}" CACHE FILEPATH "")
+endif()
+
+if(NOT CMAKE_MAKE_PROGRAM)
```

Comment:
What if `ninja` is already in the path? This would skip over that I think.

Reply:
In this case, that should be fine. The `ninja` that xcrun picks up here isn't special.

Reply:
Removed the check